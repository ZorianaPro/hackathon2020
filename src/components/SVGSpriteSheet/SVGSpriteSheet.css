.SVGSpriteSheet {
  width: 0;
  height: 0;
  display: none;
}
