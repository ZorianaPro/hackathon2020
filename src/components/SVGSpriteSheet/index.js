import React from "react";
import LocationSymbol from "../LocationSymbol";
import DateSymbol from "../DateSymbol";
import AddPersonSymbol from "../AddPersonSymbol";

import "./SVGSpriteSheet.css";
import LightningSymbol from "../LightningSymbol";

export const defaults = {
  className: "SVGSpriteSheet",
  id: "SVGSpriteSheet",
};

const SVGSpriteSheet = (props) => (
  <svg
    id={props.id || defaults.id}
    className={props.className || defaults.className}
  >
    <LocationSymbol />
    <DateSymbol />
    <AddPersonSymbol />
    <LightningSymbol />
  </svg>
);

export default SVGSpriteSheet;
