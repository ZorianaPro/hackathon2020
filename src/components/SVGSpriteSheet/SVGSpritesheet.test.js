import React from 'react';
import SVGSpriteSheet from './';
import { shallow } from 'enzyme';

describe('SVGSpriteSheet', () => {
  it('renders without crashing', () => {
    shallow(<SVGSpriteSheet />);
  });

  it('renders as expected', () => {
    expect(shallow(<SVGSpriteSheet />)).toMatchSnapshot();
  });

  describe('id', () => {
    it('uses the id given in props', () => {
      expect(shallow(<SVGSpriteSheet id="foo" />).exists('#foo')).toBe(true);
    });
  });

  describe('className', () => {
    it('uses the className given in props', () => {
      expect(shallow(<SVGSpriteSheet className="bar" />).exists('.bar')).toBe(true);
    });
  });
});
