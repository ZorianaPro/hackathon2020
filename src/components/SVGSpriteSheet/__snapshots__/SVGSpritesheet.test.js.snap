// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGSpriteSheet renders as expected 1`] = `
<svg
  className="SVGSpriteSheet"
  id="SVGSpriteSheet"
>
  <LocationSymbol />
  <DateSymbol />
</svg>
`;
